lib/utils/errors.go

```diff
@@ -75,3 +75,19 @@ func IsConnectionRefused(err error) bool {
 func IsExpiredCredentialError(err error) bool {
 	return IsHandshakeFailedError(err) || IsCertExpiredError(err) || trace.IsBadParameter(err) || trace.IsTrustError(err)
 }
+
+// IsUntrustedCertErr checks if an error is an untrusted cert error.
+func IsUntrustedCertErr(err error) bool {
+	if err == nil {
+		return false
+	}
+	errMsg := err.Error()
+	return strings.Contains(errMsg, "x509: certificate is valid for") ||
+		strings.Contains(errMsg, "certificate is not trusted")
```

Comment:
I actually discovered that the error message from untrusted certs is platform specific. On my m1 mac I was getting "...certificate is not trusted" but in CI it was getting "x509: certificate is valid for...". Either case is an untrusted certs so this checks for both in a helper function.

Reply:
Note: the error often comes back from response message and is not an x509 error type, it just contains the string. That's why I don't try using errors.As here

Reply:
Yes, this is because starting in Go 1.18, Go delegates certificate validation to the platform (OS) validator instead of implementing it in pure Go.